infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogCompaction.java

```diff
@@ -0,0 +1,63 @@
+package org.corfudb.infrastructure.log;
```

Comment:
Is all of this really necessary ? it seems like your just need to add try/catch in the runnable  task that calls stream.compact

Reply:
I believe yes, reasons: 
 - I don't want to pollute LogUnit logic 
 - I also want to measure how many times the compaction happened. I wouldn't like to put that measurement logic into LogUnit. 
 - It's easier to write a test if compaction is in a separate class.